### PR TITLE
Non-Smoke Cypress cron jenkins pipeline

### PR DESCRIPTION
**Pull request by dr3**
**Overall change:** Adds Non-Smoke Cypress cron jenkins pipeline

**Code changes:**

- Non-Smoke Cypress cron jenkins pipeline

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

**Comment by chris-hinds** [comment]
hate to be "that guy" but can I ask for "all of this" to be changed :D we are slowly in the process of moving over to a scripted pipeline syntax.

I happy to make a commit to this PR if you want me to change it.

**Comment by dr3** [comment]
@hindsc52 I have zero clue what scripted pipeline syntax is, so be my guest :P

File: Jenkinsfile-e2e [hunk 1]
```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env groovy
+
+def dockerRegistry = "329802642264.dkr.ecr.eu-west-1.amazonaws.com"
+def nodeImageVersion = "10.16.0-1"
+def nodeImage = "${dockerRegistry}/bbc-news/node-10-lts:${nodeImageVersion}"
+
+def slackChannel = '#si_repo-simorgh'
+
+// Run latest every 3 hours, between 6 & 6, Monday - Friday
+def cron_string = env.BRANCH_NAME == "latest" ? "H 0 6-6/3 * * 1-5" : ""
```
Comment (jamesbhobbs) [inline Jenkinsfile-e2e]: This should be all day every day. I would do it every 4 hours 24/7

File: Jenkinsfile-e2e [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  )
+}
+
+pipeline {
+  agent any
+  triggers {
+      cron(cron_string)
+  }
+  options {
+    timeout(time: 90, unit: 'MINUTES')
```
Comment (jamesbhobbs) [inline Jenkinsfile-e2e]:
```suggestion
    timeout(time: 239, unit: 'MINUTES')
```

Comment (jamesbhobbs) [reply Jenkinsfile-e2e]: To stop two running at once, we will fly past 90 very quickly.

File: Jenkinsfile-e2e [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                def messageParameters = [
+                  buildStatus: 'Success',
+                  branchName: env.BRANCH_NAME,
+                  colour: 'good',
+                  emoji: '🤟',
+                  slackChannel: slackChannel
+                ]
+                notifySlack(messageParameters)
+                break
+              case 'FAILRE':
```
Comment (jamesbhobbs) [inline Jenkinsfile-e2e]:
```suggestion
              case 'FAILURE':
```

**Review by jamesbhobbs — CHANGES_REQUESTED** [review]

File: Jenkinsfile-e2e (hunk 2 above)
Comment (chris-hinds) [reply Jenkinsfile-e2e]: Can you clarify this `To stop two running at once`

Do you not want the tests against different environments run in parallel?

File: Jenkinsfile-e2e (hunk 1 above)
Comment (dr3) [reply Jenkinsfile-e2e]:
```suggestion
def cron_string = env.BRANCH_NAME == "latest" ? "H 0 */4 * * *" : ""
```

File: Jenkinsfile-e2e [hunk 4]
```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env groovy
+
+def nodeImage = "329802642264.dkr.ecr.eu-west-1.amazonaws.com/bbc-news/node-10-lts:10.16.0-1"
+def slackChannel = '#si_repo-simorgh'
+
+// Run latest every 3 hours, between 6 & 6, Monday - Friday
```
Comment (dr3) [inline Jenkinsfile-e2e]:
```suggestion
// Run latest every 4 hours
```

**Comment by chris-hinds** [comment]
@benjaminhobbs @dr3 this is ready for review again now

File: Jenkinsfile-e2e [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        try {
+          stage ('Install & Setup') {
+            error('simulated error')
+            sh 'make install'
+          }
+          stage ('Running E2Es across environments') {
+            parallel (
+              'Local Prod Tests': {
+                try {
+                  sh 'CYPRESS_SMOKE=false npm run build && xvfb-run -a npm run test:prod:ci;'
```
Comment (chris-hinds) [inline Jenkinsfile-e2e:37]: Running this command from the makefile always failed to start the server `xvfb-run` would fail.

**Review by dr3 — COMMENTED** [review]
LGTM :) Thanks for the refactor chris, great work

**Review by jamesbhobbs — APPROVED** [review]

**Review by dr3 — COMMENTED** [review]
👍 Cant approve by LGTM

**Pull request merged**